ui/ui-guidelines.md

```diff
@@ -1,17 +1,60 @@
 # UI Guidelines
```

Comment:
This is my first stab at updating our UI guidelines, which hasn't been touched (besides autoformatting the markdown) since Oct 22, 2021. The project has changed a lot since then, so doing my best to reflect where we're at. I'm very open to feedback about adjusting this and would appreciate feedback!.

Reply:
Also worth nothing that git is doing some kinda confusing things with some of the diffs because of how I reorganized/reworded some of the older content. You might have an easier time reading through this using the rich diff or viewing the file.